src/numerics/include/metaphysicl/dualnumber_surrogate.h

```diff
@@ -69,9 +69,8 @@ inline DualNumberSurrogate<T, D>::DualNumberSurrogate(DualNumberSurrogate<T, D>
 }
 
 template <typename T, typename D>
-template <typename T2, typename D2>
 inline DualNumberSurrogate<T, D> &
-DualNumberSurrogate<T, D>::operator=(DualNumberSurrogate<T2, D2> & dns)
```

Comment:
Do we really need to lose the T2/D2 flexibility here?

Reply:
We're not losing it; see the next method definition. For some reason, before I had separate methods, one which took `const DualNumberSurrogate<T2,D2> &`, and another which took non-const `DualNumberSurrogate<T2,D2> &`. 

What this does is replace the extraneous `DualNumberSurrogate<T2,D2> &` with a user-defined copy assignment operator

Reply:
Ah, I see now.  Thanks!